### PR TITLE
Add a system tool to get/set hostname

### PR DESCRIPTION
**Pull request by normanr**
## Summary
Add a tool to get/set hostname

## Impact
Userspace can now get and set the system hostname

## Testing
Tested by setting hostname, and getting it back. Checked that the hostname was used in dhcp requests.

### CODE REVIEW

File: system/hostname/hostname_main.c [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+
+  if (ret == OK)
+    {
+      *val = ifr.ifr_mii_val_out;
+    }
+
+  return ret;
+}
+
+int set_phy_reg(uint16_t phy_id, uint16_t reg_num, uint16_t val)
```
Comment (xiaoxiang781216) [inline system/hostname/hostname_main.c:123]: should we remove all unrelated function and header file?

Comment (normanr) [reply system/hostname/hostname_main.c]: Woops, left over from mdio_main that I copied as a template. Removed.

**Review by xiaoxiang781216 — APPROVED** [review]
LGTM.

**Comment by normanr** [comment]
fwiw, `uname -n` can also be used to get the hostname, but it's not as intuitive, and doesn't support setting the hostname (which inspired adding the tool).

**Comment by normanr** [comment]
A related question: I want to automatically include a unique identifier from the board, eg network mac address or something, in the hostname. What do you think the best way to achieve that would be? Obviously writing a small userspace tool to generate the hostname which can be fed to the `hostname` tool via back quotes would work, but maybe an option to append `BOARDCTL_UNIQUEID` to `LIBC_HOSTNAME` would be viable? I'd prefer to avoid it in `board_app_initialize`. (Also I noticed `NETUTILS_DHCPC_HOST_NAME`, maybe it should just be replace with `LIBC_HOSTNAME` these days?)

**Comment by xiaoxiang781216** [comment]
> fwiw, `uname -n` can also be used to get the hostname, but it's not as intuitive, and doesn't support setting the hostname (which inspired adding the tool).

It's fine to add hostname program since it exists on FreeBSD/Linux too.


> A related question: I want to automatically include a unique identifier from the board, eg network mac address or something, in the hostname. What do you think the best way to achieve that would be? Obviously writing a small userspace tool to generate the hostname which can be fed to the `hostname` tool via back quotes would work, but maybe an option to append `BOARDCTL_UNIQUEID` to `LIBC_HOSTNAME` would be viable? 

you can create a PR and see the feedback from other.

> I'd prefer to avoid it in `board_app_initialize`. (Also I noticed `NETUTILS_DHCPC_HOST_NAME`, maybe it should just be replace with `LIBC_HOSTNAME` these days?)

or let `NETUTILS_DHCPC_HOST_NAME` default value to `LIBC_HOSTNAME`.

**Pull request merged**